src/chromatix/utils/data.py

```diff
@@ -22,3 +23,160 @@ def siemens_star(num_pixels=512, num_spokes=32):
             S[in_spoke] = 1.0
     S *= R < (num_pixels / 2.0)
     return S
+
+
+def draw_disks(image_size, coordinates, radius, color=255):
+    """
+    Create a grayscale image with disks drawn at each provided coordinate.
+
+    Args:
+        image_size (tuple): The desired image size as (height, width).
+        coordinates (list or numpy.ndarray): A list of (x, y) coordinates where disks should be drawn.
+        radius (int): The radius of the disks.
+        color (int): An optional intensity for the disks (0-255).
+
+    Returns:
+        numpy.ndarray: The resulting grayscale image with disks drawn at the specified coordinates.
+    """
+    # Create a blank grayscale image with the desired size
+    image = np.zeros(image_size, dtype=np.uint8)
+
+    # Draw a disk at each coordinate
+    for coord in coordinates:
+        cv2.circle(image, (coord[0], coord[1]), radius, color, -1)
+
+    return image
+
+class RandDiskGenerator: # TODO avoid overlapping disks
+    def __init__(self,
+                 N,
+                 n_points,
+                 radius,
+                 shape,
+                 z_range):
+        '''
+        Create a dataset of random 3D coordinates and their associated image.
```

Comment:
We need to redo the docstrings here to match the rest, but I can take care of that myself. I would also rename some of these variables: `n_points -> num_points`, `image_size -> shape`.